### PR TITLE
jQuery-less blockUI (almost) [POC]

### PR DESCRIPTION
**Pull request by apenchev**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

As described in #36604.

### How to test the changes in this Pull Request:

I've been testing this by simply adding items to the cart (via AJAX), removing items from the cart (via AJAX) and clicking around the checkout page.

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by samueljseay** [comment]
@apenchev I love this, thank you!

The only thing I'm thinking about is, recently, we decided to warn users of deprecation of unused scripts such as in https://github.com/woocommerce/woocommerce/pull/36576 and I'm thinking an approach like this might make sense for us in this case too. We could warn users that we won't enqueue blockUI in future versions. But before we make that change I want to have some discussion with some members of my team about this approach.

My reason for being a bit apprehensive about this change is, if other plugins are relying on blockUI its really hard to know what impact major changes to the code might do and it'd be simpler to just tell them we won't support it much longer. (Which seems reasonable considering how you've replaced all its used functionality very simply).

TLDR; give me some time to have a discussion with my team about this approach and I'll come back to you. Thanks again for your contribution.

**Comment by apenchev** [comment]
> We could warn users that we won't enqueue blockUI in future versions. But before we make that change I want to have some discussion with some members of my team about this approach.

Yes, something like this makes the most sense to me as well. The Revolut Payment Gateway plugin definitely relies on jQuery BlockUI being loaded, for instance. Although it does work with the BlockUI in this PR, it's not hard to imagine that other plugins that rely on jQuery BlockUI need some of its other functionality.

**Comment by samueljseay** [comment]
@apenchev I talked to the team and indeed I think we'll go forward with deprecation and removal instead. 

I think we'll aim for a time frame of 6 months to give external teams time to move off this script. I just need to finish my work on adding warnings for deprecated scripts.

So I think for now we should close this PR, but as usual I really appreciate your contributions!
